thriftpy/thrift.py

```diff
@@ -231,16 +231,65 @@ def process(self, iprot, oprot):
         api, seqid, result, call = self.process_in(iprot)
 
         if isinstance(result, TApplicationException):
-            self.send_exception(oprot, api, result, seqid)
-        else:
-            try:
-                result.success = call()
-            except Exception as e:
-                # raise if api don't have throws
-                self.handle_exception(e, result)
-
-            if not result.oneway:
-                self.send_result(oprot, api, result, seqid)
+            return self.send_exception(oprot, api, result, seqid)
+
+        try:
+            result.success = call()
+        except Exception as e:
+            # raise if api don't have throws
+            self.handle_exception(e, result)
+
+        if not result.oneway:
+            self.send_result(oprot, api, result, seqid)
+
+
+class TMultiplexingProcessor(TProcessor):
+    processors = {}
+    service_map = {}
+
+    def __init__(self):
+        pass
+
+    def register_processor(self, processor):
+        service = processor._service
+        name = service.__name__
```

Comment:
It may be better to use a more unique name to avoid collision of services with the same name but implementing complimentary methods. Maybe the module name and the service class name?